Adds elgg_extract_class() and fixes #10102

mrclay opened the pull request:
**chore(views): list $vars no longer pollute the image block wrapper**

In #9944 `elgg_view_image_block()` started rendering arbitrary $vars as attributes. The problem is that `object/elements/summary` also passed all $vars into it from the list view. This limits the $params passed in from `object/elements/summary`.

Fixes #10102

**feature(views): adds function for extracting $vars['class'] more cleanly**

This cleans up the pattern of creating an array of classnames by extracting `$vars['class']` and optionally merging other classnames.

Adds missing tests for `elgg_extract()`.

mrclay committed:
chore(views): list $vars no longer pollute the image block wrapper

In #9944 `elgg_view_image_block()` started rendering arbitrary $vars as
attributes. The problem is that `object/elements/summary` also passed all
$vars into it from the list view. This limits the $params passed in
from `object/elements/summary`.

Fixes #10102

hypeJunction commented:
I had a reason for this but don't remeber it anymore.  LGTM

mrclay commented on views/default/object/elements/summary.php:
```diff
@@ -44,15 +44,12 @@
-	$params['class'] = $class;
-
-	$params['data-guid'] = $entity->guid;
+	$params = [
```
Should we build this from `elgg_extract('image_block_attrs', $vars, [])`? That would leave the ability to specify it.

hypeJunction replied:
I think it's fine for now. I don't think we will be passing image block vars from the elgg_list_entities(). In 3.0, we can just wrap it in a proper view.

mrclay committed:
feature(views): adds function for extracting $vars['class'] more cleanly

This cleans up the pattern of creating an array of classnames by extracting
`$vars['class']` and optionally merging other classnames.

Adds missing tests for `elgg_extract()`.

mrclay commented:
@hypeJunction I've refreshed this completely

hypeJunction commented:
LGTM
